src/main/scala/cromwell/util/google/GoogleCredentialFactory.scala

```diff
@@ -114,4 +116,14 @@ abstract class GoogleCredentialFactory {
       case unrecognized => Failure(new IllegalArgumentException(s"Unrecognized userSchemeAuthentication: $unrecognized"))
     } getOrElse Failure(new Throwable("No user authentication configuration has been found."))
   }
+
+  private def forApplicationDefaultCredentials(): Credential = {
+    try {
+      GoogleCredential.getApplicationDefault().createScoped(GoogleScopes.Scopes.asJava)
+    } catch {
+      case e : IOException =>
+      log.warn("Failed to get application default credentials", e)
+      throw e
```

Comment:
I wasn't seeing the error until I threw, but perhaps that was a mistake on
my part.

On Wed, Jan 6, 2016 at 2:51 PM, mcovarr <notifications@github.com> wrote:

> In src/main/scala/cromwell/util/google/GoogleCredentialFactory.scala
> <https://github.com/broadinstitute/cromwell/pull/356#discussion_r49001095>
> :
>
> > @@ -114,4 +116,14 @@ abstract class GoogleCredentialFactory {
> >        case unrecognized => Failure(new IllegalArgumentException(s"Unrecognized userSchemeAuthentication: $unrecognized"))
> >      } getOrElse Failure(new Throwable("No user authentication configuration has been found."))
> >    }
> > +
> > +  private def forApplicationDefaultCredentials(): Credential = {
> > +    try {
> > +      GoogleCredential.getApplicationDefault().createScoped(GoogleScopes.Scopes.asJava)
> > +    } catch {
> > +      case e : IOException =>
> > +      log.warn("Failed to get application default credentials", e)
> > +      throw e
>
> hrm yeah, a pattern of log + throw will result in multiple copies of the
> same error / stacktrace in the logs
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/cromwell/pull/356/files#r49001095>.
>